pkg/controller/jobs/deployment/replicaset/rs_controller.go

```diff
@@ -0,0 +1,208 @@
+
+// IsSuspended always returns false since ReplicaSet cannot be suspended.
+func (r *ReplicaSet) IsSuspended() bool {
+	return r.suspended
+}
+
+// IsActive always returns true since ReplicaSet has no inactive state distinct from suspension.
+func (r *ReplicaSet) IsActive() bool {
+	return !r.suspended
+}
+
+// Suspend is a no-op for ReplicaSet, which does not support suspension.
+func (r *ReplicaSet) Suspend() {
+	r.suspended = true
+}
+
+// GVK returns apps/v1, Kind=ReplicaSet.
+func (r *ReplicaSet) GVK() schema.GroupVersionKind { return GroupVersionKind }
+
+// PodLabelSelector returns the string form of the ReplicaSet's label selector.
+// Returns an empty string when the selector is nil.
+func (r *ReplicaSet) PodLabelSelector() string {
+	if r.Spec.Selector == nil {
+		return ""
+	}
+	return r.Spec.Selector.String()
+}
+
+// PodSets exposes the ReplicaSet as a single PodSet with the template and desired replica count.
+func (r *ReplicaSet) PodSets(_ context.Context) ([]kueue.PodSet, error) {
+	podSet := kueue.PodSet{
+		Name:     kueue.DefaultPodSetName,
+		Template: *r.Spec.Template.DeepCopy(),
+		Count:    ptr.Deref(r.Spec.Replicas, 0),
+	}
+	return []kueue.PodSet{podSet}, nil
+}
+
+// RunWithPodSetsInfo mutates the ReplicaSet Pod template with admission-time adjustments
+// like node selectors, tolerations, and annotations. If Count is provided, it also updates
+// .spec.replicas to match.
+func (r *ReplicaSet) RunWithPodSetsInfo(_ context.Context, podSetsInfo []podset.PodSetInfo) error {
+	if len(podSetsInfo) != 1 {
+		return podset.BadPodSetsInfoLenError(1, len(podSetsInfo))
+	}
+	if err := podset.Merge(&r.Spec.Template.ObjectMeta, &r.Spec.Template.Spec, podSetsInfo[0]); err != nil {
+		return err
+	}
+	// Keep desired replicas in sync with the admitted PodSet count.
+	if r.Spec.Replicas == nil || *r.Spec.Replicas != podSetsInfo[0].Count {
+		r.Spec.Replicas = ptr.To(podSetsInfo[0].Count)
+	}
+	return nil
+}
+
+// RestorePodSetsInfo reverts previously merged admission-time fields, if present.
+// Returns true when a restoration occurred. Replicas are restored to the provided Count.
+func (r *ReplicaSet) RestorePodSetsInfo(podSetsInfo []podset.PodSetInfo) bool {
+	if len(podSetsInfo) == 0 {
+		return false
+	}
+	restored := podset.RestorePodSpec(&r.Spec.Template.ObjectMeta, &r.Spec.Template.Spec, podSetsInfo[0])
+	if restored {
+		r.Spec.Replicas = ptr.To(podSetsInfo[0].Count)
+	}
+	return restored
+}
+
+// Finished reports whether the ReplicaSet has a terminal outcome.
+// ReplicaSet is a steady-state controller, so it never "finishes".
+//
+// For ElasticJobsViaWorkloadSlices feature with workload-slicing annotation it
+// always returns finished=false; true otherwise.
+//
+// The success value is true to indicate no failure state.
+func (r *ReplicaSet) Finished(_ context.Context) (message string, success, finished bool) {
+	return "", true, !features.Enabled(features.ElasticJobsViaWorkloadSlices) || !workloadslicing.Enabled(r)
```

Comment:
ReplicaSet is used for serving and for serving workloads we never answer this as True: https://github.com/kubernetes-sigs/kueue/blob/d63846c9ef4c4c8244996e6f72b729b4ead6a0a4/pkg/controller/jobs/pod/pod_controller.go#L338-L340